src/arch/mod.rs

```diff
@@ -345,3 +345,31 @@ pub mod mie {
     pub const LCOFIE_OFFSET: usize = 13;
     pub const LCOFIE_FILTER: usize = 0b1 << LCOFIE_OFFSET;
 }
+
+// ———————————————————— Machine Trap-Vector Base-Address ———————————————————— //
+
+#[allow(unused)]
+pub mod mtvec {
+    /// Constant to filter out MODE bits of mtvec
+    pub const MODE_FILTER: usize = 0b11;
+
+    /// Constant to filter out BASE bits of mtvec
+    pub const BASE_FILTER: usize = usize::MAX - MODE_FILTER;
```

Comment:
Si indeed that does work, but it is kind of unusual to do bit manipulations with subtractions. In particular it works because all of the bits of `usize::MAX` are one, and if someone change it in the future this will break so it is kind of fragile.

What I would use here is the usual `usize::MAX & !MODE_FILTER` which does the same thing but is a bit more robust and commonly used.